src/modules/apps/update.ts

```diff
@@ -1,67 +1,90 @@
-  return app
-}
+const printAppsDiff = (resolvedUpdates: any, message: string, path: string[], filterFunction?: (obj: any) => boolean, pluckFunction?: (obj: any) => any) => {
+  if (!filterFunction) {
+    filterFunction = identity
+  }
+  if (!pluckFunction) {
+    pluckFunction = identity
+  }
 
-export default async () => {
-  const spinner = ora('Getting available updates').start()
-  const { data } = await listApps()
-  const installedApps = reject<Manifest>(isLinked, map(extractAppLocator, data))
-  const withLatest = await Bluebird.all(map(async (app) => {
-    app.latest = await appLatestVersion(`${app.vendor}.${app.name}`, toMajorRange(app.version))
-    return app
-  }, installedApps))
-  const updateableApps = reject(sameVersion, withLatest)
+  const appsToBeUpdated = compose<any, any, any, any>(
+    pluckFunction,
+    filterFunction,
+    ramdaPath(concat(['updates'], path))
+  )(resolvedUpdates) as string[]
+
+  const appMajorsToBeUpdated = map(toMajorRange, appsToBeUpdated)
 
-  const table = createTable({ head: ['App', 'Current', 'Latest'] })
-  updateableApps.forEach(({ vendor, name, version, latest }) => {
-    if (!latest) {
-      log.debug(`Couldn't find latest version of ${vendor}.${name}`)
-      return
-    }
-    const [fromVersion, toVersion] = diffVersions(version, latest)
+  const currentApps = compose<any, any, any, any>(
+    filter((appId: string) => includes(toMajorRange(appId), appMajorsToBeUpdated)),
+    pluckFunction,
+    ramdaPath(concat(['state'], path))
+  )(resolvedUpdates) as string[]
 
-    const formattedName = `${chalk.blue(vendor)}${chalk.gray.bold('.')}${name}`
-    table.push([formattedName, fromVersion, toVersion])
-  })
-  spinner.stop()
+  const diffTable = matchedDepsDiffTable('current', 'latest', currentApps, appsToBeUpdated)
+  if (diffTable.length === 1) {
+    return false
+  }
+  log.info(message)
+  console.log(diffTable.toString())
+  return true
+}
 
-  if (isEmpty(updateableApps)) {
-    return log.info('No updates available for installed apps.')
+const printEditionAppsDiff = (resolvedUpdates: any) => {
+  const oldState = ramdaPath(['state', 'edition'], resolvedUpdates) as string[]
+  const newState = difference(
+    union(oldState, ramdaPath(['updates', 'edition', 'install'], resolvedUpdates)),
+    ramdaPath(['updates', 'edition', 'uninstall'], resolvedUpdates)
+  )
+  const diffTable = matchedDepsDiffTable('current', 'latest', oldState, newState)
+  if (diffTable.length === 1) {
+    return false
   }
+  log.info(`The following apps will be installed/uninstalled due to changes to current edition`)
+  console.log(diffTable.toString())
+  return true
+}
 
-  console.log(`${table.toString()}\n`)
+export default async () => {
+  const housekeeper = new Housekeeper(getIOContext(), IOClientOptions)
+  const getSpinner = ora('Getting available updates').start()
+  const resolvedUpdates = await housekeeper.resolve()
+  getSpinner.stop()
+  const hasAvailableUpdates =
```

Comment:
The names of `print` methods are suggesting they only have to print something, but actually they also are returning a boolean when there are available updates.
What do you think about separate these two things: print and check available updates?  `hasAvailableUpdates` can be only a check on each field resolvedUpdates to check if the length isn't zero.
 If there are availableUpdates you can can a method `printResolvedUpdates`. So this one will run the four `printAppsDiff`.